src/envoy/http/backend_auth/filter.cc

```diff
@@ -53,6 +53,9 @@ const Envoy::Http::LowerCaseString kXForwardedAuthorization{
 FilterHeadersStatus Filter::decodeHeaders(RequestHeaderMap& headers, bool) {
   // Make sure route is calculated
   auto route = decoder_callbacks_->route();
+
+  // This shouldn't happen as the fallback wildcard route match should catch all
+  // the undefined requests.
```

Comment:
Then, we should remove dead code.  and use ASSERT or  NOT_REACHABLE_CODE.

Reply:
Yes, why did we add this code back in? I thought the original commit deletes it.

Reply:
This is to explain why all the requests should be matched